src/screens/FeedScreen/hooks/useFeedPreloadHook.ts

```diff
@@ -0,0 +1,28 @@
+import {NativeScrollEvent, NativeSyntheticEvent} from 'react-native';
+
+import useFeedUtilsHook from './useFeedUtilsHook';
+import {FEEDS_DIFF_TO_FETCH} from '../../../utils/constants';
+
+const useFeedPreloadHook = (feedLength: number, fetchCallback: () => void) => {
+  const {getCurrentPostViewed} = useFeedUtilsHook();
+
+  function __shouldFetchNextFeeds(momentumEvent: NativeSyntheticEvent<NativeScrollEvent>): boolean {
+    const shownIndex = getCurrentPostViewed(momentumEvent);
+    console.log(`shown index ${shownIndex} vs ${feedLength}`);
+    return feedLength - shownIndex <= FEEDS_DIFF_TO_FETCH;
+  }
+
+  function fetchNextFeeds(momentumEvent: NativeSyntheticEvent<NativeScrollEvent>) {
+    const shouldFetch = __shouldFetchNextFeeds(momentumEvent);
+    if (shouldFetch && fetchCallback) {
+      console.log('fetch next feeds');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, logging actions like 'fetch next feeds' could potentially expose sensitive application behavior. Ensure this is only enabled in development environments.
```diff
- console.log('fetch next feeds');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->